force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,46 @@
+trigger AccountTrigger on Account (before insert, after insert) {
+        // Before Insert Logic
+    if(Trigger.isBefore && Trigger.isInsert){
+        for(Account acc : Trigger.new){
+        // Set the Account Type to 'Prospect' if it's null
+        if(acc.Type == null){
+            acc.Type = 'Prospect';
+        }
+        
+ 	    // Copy Shipping Address to Billing Address if Shipping fields are not empty
+ 	    if(acc.ShippingStreet != null || 
+           acc.ShippingCity != null ||
+           acc.ShippingState != null ||
+           acc.ShippingPostalCode != null ||
+           acc.ShippingCountry != null){
+                acc.BillingStreet = acc.ShippingStreet;
+                acc.BillingCity = acc.ShippingCity;
+                acc.BillingState = acc.ShippingState;
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+                acc.BillingCountry = acc.ShippingCountry;
+        }
+        
+        //Update rating to 'Hot' if Fax, Website and Phone all have values.
+        if(acc.Phone != null && acc.Fax != null && acc.Website != null){
+            acc.Rating = 'Hot';
+   		 }
+      }
```

Comment:
Try to separate these 3 methods into individual ones, it'll be easier to test and you can find out problems quickly.